tests/test_cli_compile.py

```diff
@@ -2582,6 +2582,30 @@ def test_input_formats(fake_dists, runner, make_module, fname, content):
     assert "extra ==" not in out.stderr
 
 
+def test_error_in_pyproject_toml(fake_dists, runner, make_module, capfd):
+    """
+    Test that an error in pyproject.toml is reported.
+    """
+    fname = "pyproject.toml"
+    invalid_content = dedent(
+        """\
+            [project]
+            invalid = "metadata"
+            """
```

Comment:
```suggestion
        """\
        [project]
        invalid = "metadata"
        """
```

Shifted indent slightly to fit the line.